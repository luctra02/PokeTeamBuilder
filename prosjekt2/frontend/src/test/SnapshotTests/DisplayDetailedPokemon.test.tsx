import { screen, waitFor } from '@testing-library/react';
import Details from '../../components/DetailsCardComponent';
import { allMocks as mocks} from '../mocks/mockQueries';
import { getPokemonsMock } from '../mocks/mockQueries';
import { testPageRender } from '../testUtils';





describe('renders a pokemon', () => {
    it('matches the snapshot', async () => {
      const { asFragment } = testPageRender(<Details pokemon={getPokemonsMock.result.data.getPokemons.pokemons[0]} />, { mocks });  
      await waitFor(() => {
 
        expect(screen.getByText(/pikachu/)).toBeInTheDocument();
        expect(screen.getByText(/Weight: 6 kg/)).toBeInTheDocument();
        expect(screen.getByText(/Height: 0.4 m/)).toBeInTheDocument();
        expect(screen.getByText(/Base Stats/))

      });
      expect(asFragment()).toMatchSnapshot();
    });
  
    /* it('test add to team button', async () => {
      testPageRender(<Details pokemon={getPokemonsMock.result.data.getPokemons.pokemons[0]} />, { mocks });
      const addToTeamButton = await screen.findByText(/Add to Team/);
      userEvent.click(addToTeamButton);
  
      await waitFor(() => {
        expect(screen.getByText(/Remove from Team/)).toBeInTheDocument();
      });
    }); */
  });
