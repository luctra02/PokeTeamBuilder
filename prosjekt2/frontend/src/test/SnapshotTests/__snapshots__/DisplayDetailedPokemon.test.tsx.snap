// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders a pokemon > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="pokemonInfo"
  >
    <h1>
      #25 pikachu
    </h1>
    <div
      class="pokemonInfoWrapper"
    >
      <div
        class="pokeImage"
      >
        <img
          alt=""
          src="https://play.pokemonshowdown.com/sprites/ani/pikachu.gif"
        />
      </div>
      <div
        class="pokeStats"
      >
        <div
          class="bodyStats"
        >
          <p>
            Weight: 6 kg
          </p>
          <p>
            Height: 0.4 m
          </p>
        </div>
        <div
          class="pokeType"
        >
          <div>
            <div
              class="pokeType PokeTypeBox"
              style="background-color: rgb(248, 208, 48);"
            >
              Electric
            </div>
          </div>
        </div>
        <div
          class="baseStats"
        >
          <h3>
            Base Stats
          </h3>
          <div
            class="stat-container"
          >
            <div
              class="fill-bar-container"
              id="attackFill"
            >
              <div
                class="fill-bar"
                style="height: 30px;"
              />
            </div>
            <label
              for="attackFill"
            >
              attack
            </label>
          </div>
          <div
            class="stat-container"
          >
            <div
              class="fill-bar-container"
              id="defenseFill"
            >
              <div
                class="fill-bar"
                style="height: 25px;"
              />
            </div>
            <label
              for="defenseFill"
            >
              defense
            </label>
          </div>
          <div
            class="stat-container"
          >
            <div
              class="fill-bar-container"
              id="hpFill"
            >
              <div
                class="fill-bar"
                style="height: 20px;"
              />
            </div>
            <label
              for="hpFill"
            >
              hp
            </label>
          </div>
          <div
            class="stat-container"
          >
            <div
              class="fill-bar-container"
              id="speedFill"
            >
              <div
                class="fill-bar"
                style="height: 55px;"
              />
            </div>
            <label
              for="speedFill"
            >
              speed
            </label>
          </div>
          <div
            class="stat-container"
          >
            <div
              class="fill-bar-container"
              id="specialAttackFill"
            >
              <div
                class="fill-bar"
                style="height: 30px;"
              />
            </div>
            <label
              for="specialAttackFill"
            >
              special attack
            </label>
          </div>
          <div
            class="stat-container"
          >
            <div
              class="fill-bar-container"
              id="specialDefenseFill"
            >
              <div
                class="fill-bar"
                style="height: 25px;"
              />
            </div>
            <label
              for="specialDefenseFill"
            >
              special defense
            </label>
          </div>
        </div>
      </div>
    </div>
    <button
      class="teamButton"
    >
      Add to Team
    </button>
  </div>
</DocumentFragment>
`;
